packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -144,3 +145,12 @@ export function isVFragment(vnode: VNode): vnode is VFragment {
 export function isVScopedSlotFragment(vnode: VNode): vnode is VScopedSlotFragment {
     return vnode.type === VNodeType.ScopedSlotFragment;
 }
+
+export function areDynamicVCustomElements(vnode1: VNode, vnode2: VNode): boolean {
+    return (
+        vnode1.type === VNodeType.CustomElement &&
+        vnode1.dynamic &&
```

Comment:
The `dynamic` field is only used for this check. Based on the other discussion about marking dynamic components as dynamic for the purpose of diffing, this field can be removed.